osu.Game/OsuGame.cs

```diff
@@ -249,7 +249,7 @@ private void unregisterBlockingOverlay(OverlayContainer overlayContainer)
         /// <param name="hideToolbar">Whether the toolbar should also be hidden.</param>
         public void CloseAllOverlays(bool hideToolbar = true)
         {
-            foreach (var overlay in focusedOverlays)
+            foreach (var overlay in focusedOverlays.ToArray())
```

Comment:
Isn't this just moving the problem somewhere else? If the collection is modified during the array copy, it'll fail still, albeit with a slightly different error.

If there is thread contention on this collection then the correct thing to do is probably either to schedule all mutations of the list onto one thread, or add locking.

Reply:
Ah? I figured it was actually one of the `Hide` calls causing an unregister.

Didn't see any case of threading involved.

Reply:
The unregister can - in theory - come from async disposal, and in all current usages this seems to be the case. Example:

https://github.com/ppy/osu/blob/d43b980a172cf4b6a3a78e3e5f32d516998a20e9/osu.Game/Screens/Select/SongSelect.cs#L705

Which I guess is my bad since I wrote that mechanism and neglected to spot that.

Reply:
I've `Schedule`d the unregister instead.